src/server/pps/server/api_server_test.go

```diff
@@ -524,6 +528,103 @@ func TestSetClusterDefaults(t *testing.T) {
+	pipeline := tu.UniqueString("pipeline")
+
+	// commit data to the repo so there will be datums to process
+	require.NoError(t, c.CreateRepo(pfs.DefaultProjectName, repo))
+	commit1, err := c.StartCommit(pfs.DefaultProjectName, repo, "master")
+	require.NoError(t, err)
+	require.NoError(t, c.PutFile(commit1, "/file.txt", &bytes.Buffer{}))
+	require.NoError(t, c.FinishCommit(pfs.DefaultProjectName, repo, "master", commit1.Id))
+	_, err = c.WaitCommit(pfs.DefaultProjectName, repo, "master", commit1.Id)
+	require.NoError(t, err)
+
+	// create the pipelne
+	spec := &pps.CreatePipelineRequest{
+		Pipeline: &pps.Pipeline{
+			Project: &pfs.Project{
+				Name: pfs.DefaultProjectName,
+			},
+			Name: pipeline,
+		},
+		Transform: &pps.Transform{Cmd: []string{"bash"}, Stdin: []string{"date > /pfs/out/date.txt"}},
+		Input: &pps.Input{
+			Pfs: &pps.PFSInput{
+				Repo: repo,
+				Glob: "/",
+			},
+		},
+	}
+	js, err := protojson.Marshal(spec)
+	require.NoError(t, err, "marshalling JSON must not error")
+	_, err = c.PpsAPIClient.CreatePipelineV2(ctx, &pps.CreatePipelineV2Request{
+		CreatePipelineRequestJson: string(js),
+	})
+	require.NoError(t, err, "CreatePipelineV2 must succeed")
+	// wait for job to finish and get output commit data
+	jobs, err := c.ListJob(pfs.DefaultProjectName, pipeline, nil, 0, true)
+	require.NoError(t, err)
+	require.Equal(t, 1, len(jobs))
+	_, err = c.WaitJob(pfs.DefaultProjectName, pipeline, jobs[0].Job.GetId(), false)
+	require.NoError(t, err)
+	var date1 bytes.Buffer
+	require.NoError(t, c.GetFile(client.NewCommit(pfs.DefaultProjectName, pipeline, "master", ""), "/date.txt", &date1))
+	require.NotEqual(t, date1.String(), "")
+
+	// rerun pipeline
+	_, err = c.PpsAPIClient.RerunPipeline(ctx, &pps.RerunPipelineRequest{
+		Pipeline: &pps.Pipeline{
+			Project: &pfs.Project{Name: pfs.DefaultProjectName},
+			Name:    pipeline,
+		},
+	})
+	require.NoError(t, err, "RerunPipeline must succeed")
+	r, err := c.PpsAPIClient.InspectPipeline(ctx, &pps.InspectPipelineRequest{Pipeline: &pps.Pipeline{Name: pipeline}})
+	require.NoError(t, err, "InspectPipeline must succeed")
+	require.Equal(t, r.Version, uint64(2), "pipeline version should = 2")
+	// wait for job to finish and get output commit data
+	jobs, err = c.ListJob(pfs.DefaultProjectName, pipeline, nil, 0, true)
+	require.NoError(t, err)
+	require.Equal(t, 1, len(jobs))
+	_, err = c.WaitJob(pfs.DefaultProjectName, pipeline, jobs[0].Job.GetId(), false)
+	require.NoError(t, err)
+	var date2 bytes.Buffer
+	require.NoError(t, c.GetFile(client.NewCommit(pfs.DefaultProjectName, pipeline, "master", ""), "/date.txt", &date2))
+	require.NotEqual(t, date2.String(), "")
+	require.Equal(t, date1.String(), date2.String())
+
+	// rerun pipeline with reprocess
+	_, err = c.PpsAPIClient.RerunPipeline(ctx, &pps.RerunPipelineRequest{
+		Pipeline: &pps.Pipeline{
+			Project: &pfs.Project{Name: pfs.DefaultProjectName},
+			Name:    pipeline,
+		},
+		Reprocess: true,
```

Comment:
this could more explicitly test that the datums are reprocessed by inspecting the resulting Job's stats